server/routes/teams.ts

```diff
@@ -67,12 +67,14 @@ export default function routes({
       const channelTree = monitoringChannelService.generateChannelTree(channelRecommendations)
 
       // Check for legacy channels
-      const hasLegacyChannels = channelRecommendations.recommendations.some(
+      const legacyChannelCount = channelRecommendations.recommendations.filter(
         rec =>
           rec.currentChannels.dev === '#dps_alerts_non_prod' ||
           rec.currentChannels.preprod === '#dps_alerts_non_prod' ||
           rec.currentChannels.prod === '#dps_alerts',
-      )
+      ).length
+
+      const hasLegacyChannels = legacyChannelCount > 0
```

Comment:
Genuinely, what is the difference here?

`.some()` returns `false` if none are found, otherwise `true`. `filter().length` returns 0 if none are found and > 0 if some are found. This check then checks the count > zero?

Or am I misunderstanding `.some()`?